### PR TITLE
Added Quzar's VMUBeep as an Example

### PR DESCRIPTION
**Pull request by gyrovorbis**
@QuzarDC's VMUBeep is an incredibly useful tool that allows you to easily send raw maple buzzer commands to a VMU. It was just able to help us do some R&D on how the buzzer behaves with the 6Mhz VMU clock versus the slower ones used in standalone mode.

- integrated the tool into KOS as an example
- modifified a bunch of the comments, now that we actually know what these raw bytes mean
- trivial formatting changes
- uintN types -> uintN_t types
- removed warnings

### CODE REVIEW

**Review by ljsebald — CHANGES_REQUESTED** [review]
A couple of things I'd like to see fixed before merging this:

1. Fix the Copyright notices to be correct. They should be of the following form, technically speaking: `Copyright (C) Year Author` (and only one author per line)
2. Fix the Makefile for the example to match other examples (i.e, using `kos-cc` instead of `$(KOS_CC)` and other such things
3. Change Tabs to 4 spaces and generally fix formatting in the code (for instance, the curly brace on the `for(;;)` loop, spaces after `while` and `if`, missing spaces between `)` and `{` on some conditionals, missing spaces between conditional blocks, etc. Also, the formatting on the header comment needs to be fixed (there shouldn't be tabs there either).
4. It'd also be nice if the example included the header files it actually needs, rather than just including `<kos.h>`. But that's more my personal preference there...

**Comment by gyrovorbis** [comment]
Sounds good. I think got them all.

I actually thought that if multiple people contributed at the same time to a file, they were supposed to be comma separated. WHOOPS!

File: examples/dreamcast/vmu/vmu_beep/Makefile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#
+# VMUBeep Test 2
+# (C) 2008 Donald Haase
```
Comment (ljsebald) [inline examples/dreamcast/vmu/vmu_beep/Makefile:3]:
```suggestion
# Copyright (C) 2008 Donald Haase

```

File: examples/dreamcast/vmu/vmu_beep/beep.c [hunk 2]
```diff
@@ -0,0 +1,195 @@
+/*  VMU Beep for KallistiOS ##version##
+
+    beep.c
+    (C) 2004 SinisterTengu
+    (C) 2008 Donald Haase
+
+*/
```
Comment (ljsebald) [inline examples/dreamcast/vmu/vmu_beep/beep.c:7]:
```suggestion
/*  KallistiOS ##version##

    beep.c
    Copyright (C) 2004 SinisterTengu
    Copyright (C) 2008 Donald Haase

*/
```

File: examples/dreamcast/vmu/vmu_beep/beep.c [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    point_t w;
+    int i = 0, count = 0;
+    uint16_t old_buttons = 0, rel_buttons = 0;
+    uint32_t effect = 0;
+    uint8_t n[8] = { 0, 0, 0, 0, 0, 0, 0, 0 }; //nibbles
+    char s[8][2] = { "", "", "", "", "", "", "", "" };
+
+    /*if these buttons are pressed, exit_app */
+    cont_btn_callback(0, CONT_START | CONT_A | CONT_B | CONT_X | CONT_Y,
+        (void(*)(uint8_t, uint32_t))arch_exit);
```
Comment (ljsebald) [inline examples/dreamcast/vmu/vmu_beep/beep.c:53]:
```suggestion
    /* If the face buttons are all pressed, exit the app */
    cont_btn_callback(0, CONT_START | CONT_A | CONT_B | CONT_X | CONT_Y,
                      (void(*)(uint8_t, uint32_t))arch_exit);
```

File: examples/dreamcast/vmu/vmu_beep/beep.c [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+    pvr_init_defaults();
+
+    fnt = plx_font_load("/rd/axaxax.txf");
+    cxt = plx_fcxt_create(fnt, PVR_LIST_TR_POLY);
+
+    pvr_set_bg_color(0.0f, 0.0f, 0.0f);
+
+    for (;;) {
```
Comment (ljsebald) [inline examples/dreamcast/vmu/vmu_beep/beep.c:62]: I meant remove the spaces, not add them.
```suggestion
    for(;;) {
```

File: examples/dreamcast/vmu/vmu_beep/beep.c [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+        dev = maple_enum_type(0, MAPLE_FUNC_CONTROLLER);
+        vmudev = maple_enum_type(0, MAPLE_FUNC_CLOCK);
+
+        while (!dev) {
```
Comment (ljsebald) [inline examples/dreamcast/vmu/vmu_beep/beep.c:66]:
```suggestion
        while(!dev) {
```

File: examples/dreamcast/vmu/vmu_beep/beep.c [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+            plx_fcxt_setpos_pnt(cxt, &w);
+            plx_fcxt_draw(cxt, "Please attach a controller!");
+            plx_fcxt_end(cxt);
+
+            pvr_scene_finish();
+
+            dev = maple_enum_type(0, MAPLE_FUNC_CONTROLLER);
+        }
+
+        while (!vmudev) {
```
Comment (ljsebald) [inline examples/dreamcast/vmu/vmu_beep/beep.c:83]:
```suggestion
        while(!vmudev) {
```

File: examples/dreamcast/vmu/vmu_beep/beep.c [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+        plx_fcxt_setpos_pnt(cxt, &w);
+        plx_fcxt_setsize(cxt, 30.0f);
+        plx_fcxt_draw(cxt, "0x");
+
+        w.x += 48.0f;
+        plx_fcxt_setpos_pnt(cxt, &w);
+
+        count = 0;
+        while (count <= 7) {
+            if (i == count)
```
Comment (ljsebald) [inline examples/dreamcast/vmu/vmu_beep/beep.c:120]:
```suggestion
        while(count <= 7) {
            if(i == count)
```

File: examples/dreamcast/vmu/vmu_beep/beep.c [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+            effect = (n[0] << 28) + (n[1] << 24) + (n[2] << 20) + (n[3] << 16) +
+                     (n[4] << 12) + (n[5] << 8) + (n[6] << 4) + (n[7] << 0);
+
+            vmu_beep_raw(vmudev, effect);
+            printf("VMU Beep: 0x%lx!\n", effect);
+        }
+
+        if ((state->buttons & CONT_B) && (rel_buttons & CONT_B)) {
+            vmu_beep_raw(vmudev, 0x00000000);
+            printf("Beep Stopped!\n");
```
Comment (ljsebald) [inline examples/dreamcast/vmu/vmu_beep/beep.c:188]:
```suggestion
        if((state->buttons & CONT_DPAD_LEFT) && (rel_buttons & CONT_DPAD_LEFT)) {
            if (i > 0) i--;
        }

        if((state->buttons & CONT_DPAD_RIGHT) && (rel_buttons & CONT_DPAD_RIGHT)) {
            if (i < 7) i++;
        }

        if((state->buttons & CONT_DPAD_UP) && (rel_buttons & CONT_DPAD_UP)) {
            if (n[i] < 15) n[i]++;
        }

        if((state->buttons & CONT_DPAD_DOWN) && (rel_buttons & CONT_DPAD_DOWN)) {
            if (n[i] > 0) n[i]--;
        }

        if((state->buttons & CONT_A) && (rel_buttons & CONT_A)) {
            effect = (n[0] << 28) + (n[1] << 24) + (n[2] << 20) + (n[3] << 16) +
                     (n[4] << 12) + (n[5] << 8) + (n[6] << 4) + (n[7] << 0);

            vmu_beep_raw(vmudev, effect);
            printf("VMU Beep: 0x%lx!\n", effect);
        }

        if((state->buttons & CONT_B) && (rel_buttons & CONT_B)) {
            vmu_beep_raw(vmudev, 0x00000000);
            printf("Beep Stopped!\n");
```

**Review by ljsebald — CHANGES_REQUESTED** [review]

**Comment by gyrovorbis** [comment]
Give me a sec to verify the example is still good to go.

**Comment by gyrovorbis** [comment]
Everything is fine here! Built and tested!

File: examples/dreamcast/vmu/vmu_beep/beep.c [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+        plx_fcxt_draw(cxt, "Press B to stop vmu beep.");
+        w.y += 25.0f;
+
+        plx_fcxt_setpos_pnt(cxt, &w);
+        plx_fcxt_draw(cxt, "Press ABXYS to quit.");
+
+        plx_fcxt_end(cxt);
+        pvr_scene_finish();
+
+		/* Store current button states + buttons which have been released. */
```
Comment (ljsebald) [inline examples/dreamcast/vmu/vmu_beep/beep.c:158]: One last one that I just noticed...
```suggestion
        /* Store current button states + buttons which have been released. */
```

**Review by ljsebald — APPROVED** [review]

**Pull request merged**